app/app.go

```diff
@@ -720,44 +806,61 @@ func initParamsKeeper(appCodec codec.BinaryMarshaler, legacyAmino *codec.LegacyA
 	paramsKeeper.Subspace(ibchost.ModuleName)
 	paramsKeeper.Subspace(aoltypes.ModuleName)
 	paramsKeeper.Subspace(didtypes.ModuleName)
-	paramsKeeper.Subspace(tokentypes.ModuleName)
 	paramsKeeper.Subspace(burntypes.ModuleName)
 	paramsKeeper.Subspace(wasm.ModuleName)
 
 	return paramsKeeper
 }
 
-func (app *App) registerUpgradeHandlers() {
-	app.UpgradeKeeper.SetUpgradeHandler("v2.0.2", func(ctx sdk.Context, plan upgradetypes.Plan) {})
-
-	app.UpgradeKeeper.SetUpgradeHandler("v2.0.3", func(ctx sdk.Context, plan upgradetypes.Plan) {
-		// Set the min-commission-rate to 3%
-		newParams := stakingtypes.NewParams(
-			app.StakingKeeper.UnbondingTime(ctx),
-			app.StakingKeeper.MaxValidators(ctx),
-			app.StakingKeeper.MaxEntries(ctx),
-			app.StakingKeeper.HistoricalEntries(ctx),
-			app.StakingKeeper.BondDenom(ctx),
-			sdk.NewDecWithPrec(3, 2), // min-commission-rate
-		)
-		app.StakingKeeper.SetParams(ctx, newParams)
-
-		// Update the commission rate of all validators whose commission rate is smaller than min-commission-rate
-		validators := app.StakingKeeper.GetAllValidators(ctx)
-		minCommissionRate := app.StakingKeeper.GetParams(ctx).MinCommissionRate
-		for _, v := range validators {
-			if v.Commission.Rate.LT(minCommissionRate) {
-				comm, err := app.StakingKeeper.MustUpdateValidatorCommission(ctx, v, minCommissionRate)
-				if err != nil {
-					panic(err)
-				}
-				v.Commission = comm
-
-				// call the before-modification hook since we're about to update the commission
-				app.StakingKeeper.BeforeValidatorModified(ctx, v.GetOperator())
-
-				app.StakingKeeper.SetValidator(ctx, v)
-			}
+// registerUpgradeHandlers registers upgrade handlers, and sets the store loader if necessary.
+// This function must be called before sealing the BaseApp (i.e. by app.LoadLatestVersion())
+// because the storetypes loader cannot be set if BaseApp is already sealed.
+func (app *App) registerUpgradeHandlers() error {
```

Comment:
If we upgrade our mainnet to v2.0.5 with this upgrade handler, I think we must remove the upgrade handler in the `master` branch which is similar as this.